packages/babel-plugin-ember-test-metadata/get-normalized-file-path.js

```diff
@@ -9,15 +9,27 @@ const path = require('path');
 
 /**
  * Get a normalized file path, based on whether the app build is classic or with Embroider
- * @param {object} opts Babel state.file.opts which include root and filename props
+ * @param {Object} opts Babel state.file.opts which include root and filename props
+ * @param {string} opts.packageName the name of the package specified in babel plugin options
+ * @param {boolean} opts.isUsingEmbroider flag for embroider builds as specified in babel plugin options
+ * @param {string} opts.filename absolute path the name of the file being visited
+ * @param {string} opts.root path to the project root (defaults to cwd)
+ * @param {string} opts.sourceFileName path.basename(opts.filenameRelative) when available, or "unknown"
  * @returns {string} E.g. tests/acceptance/my-test.js
  */
-function getNormalizedFilePath({ packageName, isUsingEmbroider, filename, root }) {
+function getNormalizedFilePath({ packageName, isUsingEmbroider, filename, root, sourceFileName }) {
   if (!isUsingEmbroider) {
     if (filename.includes('ember-add-in-repo-tests')) {
       return _getRelativePathForClassicInRepo(filename);
     }
 
+    if (sourceFileName &&
+      !path.isAbsolute(sourceFileName) &&
+      !sourceFileName.startsWith(packageName) &&
+      sourceFileName !== 'unknown') {
+      return sourceFileName;
+    }
```

Comment:
Can we make a comment to identify what specifically what case we're trying to handle here?

Reply:
Both would be preferable. People are going to go to the implementation first, and only when that implementation isn't clear will they usually resort to looking at the test cases. 

If you update the test case titles too, which would be preferred, it'd be useful to know what use case this is handling rather than articulating the implementation details.